demo-app/src/pages/platform-devices/index.jsx

```diff
@@ -0,0 +1,223 @@
+          feature.properties.geo?.state,
+          feature.properties.geo?.country,
+        ]
+          .filter(Boolean)
+          .join(', '),
+      },
+    };
+  }),
+};
+
+const devicesWithLocation = {
+  ...devices,
+  features: devices.features.filter(device => {
+    if (device.geometry.coordinates[0] !== null) {
+      return device;
+    }
+  }),
+};
+
+const devicesWithoutLocation = {
+  ...devices,
+  features: devices.features.filter(device => {
+    if (device.geometry.coordinates[0] === null) {
+      return device;
+    }
+  }),
+};
+
+const properties = {
+  'properties.serial_number': {
+    label: 'Serial number',
+    search: true,
+  },
+  'properties.device_type': {
+    label: 'Device type',
+    search: true,
+  },
+  'properties.name': {
+    label: 'Name',
+    search: true,
+  },
+  'properties.ip_address': {
+    label: 'IP address',
+    search: true,
+  },
+  'properties.mac': {
+    label: 'MAC address',
+    search: true,
+  },
+  'properties.part_number': {
+    label: 'Part number',
+    search: true,
+  },
+  'properties.geo.city': {
+    label: 'City',
+    search: true,
+  },
+  'properties.geo.state': {
+    label: 'State',
+    search: true,
+  },
+  'properties.geo.country': {
+    label: 'Country',
+    search: true,
+  },
+  'properties.location': {
+    label: 'Location',
+    search: true,
+  },
+};
+
+const columns = [
+  {
+    property: 'properties.serial_number',
+    header: 'Serial number',
+    primary: true,
+  },
+  {
+    property: 'properties.device_type',
+    header: 'Device type',
+  },
+  { property: 'properties.mac', header: 'MAC address' },
+  { property: 'properties.ip_address', header: 'IP address' },
+  { property: 'properties.name', header: 'Location name' },
+  { property: 'properties.geo.city', header: 'City' },
+  { property: 'properties.geo.state', header: 'State' },
+  { property: 'properties.geo.country', header: 'Country' },
+];
+
+const noLocationMessage = `${new Intl.NumberFormat(navigator.language).format(
+  devicesWithoutLocation.features.length,
+)} devices do not have location specified. ${new Intl.NumberFormat(
+  navigator.language,
+).format(devicesWithLocation.features.length)} of ${new Intl.NumberFormat(
+  navigator.language,
+).format(
+  devicesWithLocation.features.length + devicesWithoutLocation.features.length,
+)} devices displayed.`;
+
+const PlatformDevices = () => {
+  const containerRef = React.useRef();
+  // const [loading, setLoading] = React.useState(true);
+  const [view, setView] = React.useState('map');
+
+  // React.useEffect(() => {
+  //   setTimeout(() => {
+  //     setLoading(false);
+  //   }, 1000);
+  // }, []);
+
+  return (
+    <Page fill>
+      <PageContent flex>
+        <PageHeader
+          title="Acme, Inc. Devices"
+          parent={<ReverseAnchor as={Link} label="Home" to="/" />}
+        />
+        <ContentContainer flex={false}>
+          <Data data={devicesWithLocation.features} properties={properties}>
```

Comment:
More a question re: Data but if we have `properties` defined here, do we also need DataFilter explicitly defined below as well (lines 158 - 161)

Reply:
If I understand your question correctly, yes. If we opt out of `<Data ... toolbar />` default mode and into compositional mode, then the caller is composing which properties become filters and their order.

Reply:
Gotcha -- I didn't realize that we needed both `properties` on Data and the DataFilters/DataFilter defined below. Learning more about Data and friends.

Reply:
me too